plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenFacetImpl.java

```diff
@@ -250,10 +269,30 @@ public Content put(final MavenPath path, final Payload payload)
   public Content put(final StorageTx tx, final MavenPath path, final Payload payload)
       throws IOException, InvalidContentException
   {
+    return put(tx, path, payload, new Supplier<InputStream>()
+    {
+      @Override
+      public InputStream get() {
+        try {
+          return payload.openInputStream();
+        }
+        catch (IOException e) {
+          throw Throwables.propagate(e);
+        }
+      }
+    });
+  }
+
+  private Content put(final StorageTx tx,
+                      final MavenPath path,
+                      final Payload payload,
+                      final Supplier<InputStream> inputStreamSupplier)
```

Comment:
I don't think there's much value in using Supplier<InputStream> here instead of InputStream, at least until createBlob() takes a supplier. It seems we're just wrapping something and immediately unwrapping it again.